### PR TITLE
Add development environment using dev container

### PR DESCRIPTION
**Pull request by R-HNF**
#### Changes
* Add `Dockerfile.dev` and `devcontainer.json` to use dev container on Visual Studio Code.
* Modified the `Dockerfile` to not be fixed to `amd64`.
  * I have verified the test in the arm environment using the development container on my M2 Mac and Intel Windows.
* There are some small changes.

#### Future plans
* Prepare the document that describes how to develop GatlingOperator using a dev container.
* Utilize utilize `buildx` for creating builds that support multiple platforms.

#### References
* [Dockerfile reference | Docker Documentation](https://docs.docker.com/engine/reference/builder/)

#### Notes
* I'm using `keychain` to forward the ssh-agent from my Mac to a dev container.
  * [Funtoo Keychain Project - Funtoo](https://www.funtoo.org/Funtoo:Keychain)
* Using Rosetta for amd emulation can result in slower processing, causing kube-apiserver to fail to start, and tests may not pass.
  * kubernetes/kubernetes#72163
* You can use dev container without VSCode.
  ```
  # 1. Build the container image.
  $ docker build -f Dockerfile.dev -t gatling-operator-dev-container .
  # 2. Create and login an dev container
  $ docker run \
      --name=gatling-operator-dev-container  \
      --hostname=gatling-operator-dev-container \
      --network=host \
      --mount type=bind,src=/var/run/docker.sock,dst=/var/run/docker.sock \
      --mount type=bind,src=.,dst=/workspace \
      --env IN_DEV_CONTAINER="true" \
      -ti gatling-operator-dev-container
  # 3. If you exited, login to dev container again.
  $ docker exec -ti gatling-operator-dev-container bash
  ```

### Checklist

- [ ] Tests have been added (if applicable, ie. when operator codes are added or modified)
- [ ] Relevant docs have been added or modified (if applicable, ie. when new features are added or current features are modified)

Relevant issue #88

### CODE REVIEW

**Review by gold-kou — APPROVED** [review]
I checked that some make commands work correctly.

LGTM

File: Makefile [hunk 1]
```diff
@@ -46,12 +46,13 @@ all: build
...
 
-kind-create: ## Create a kind cluster named ${KIND_CLUSTER_NAME} locally if necessary
+kind-create: ## Create a kind cluster named ${KIND_CLUSTER_NAME} locally if necessary and save the kubectl config.
 ifeq (1, $(shell kind get clusters | grep ${KIND_CLUSTER_NAME} | wc -l | tr -d ' '))
 	@echo "Cluster already exists"
 else
 	@echo "Creating Cluster"
 	kind create cluster --name ${KIND_CLUSTER_NAME} --image=kindest/node:${K8S_NODE_IMAGE} --config ${KIND_CONFIG_DIR}/cluster.yaml
+	cp -r ${HOME}/.kube .
 endif
```
Comment (itiB) [inline Makefile:56]: There is no problem with cp command running in the dev container, but I'm not happy if it runs locally.
Can you make `if` depending on whether it's in the dev container?

i think, dev container can set some environment variables from `devcontainer.json`.
ex. Create some check variable like `IN_DEV_CONTAINER` and check it.
<https://code.visualstudio.com/remote/advancedcontainers/environment-variables>

**Review by itiB — COMMENTED** [review]
Looks well:+1:
I make some suggestion comment.

File: Makefile (hunk 1 above)
Comment (R-HNF) [reply Makefile]: @itiB 
I added `IN_DEV_CONTAINER` to check whether it's in the dev container.
Thank you for your comments.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	"mounts": [
+		// Mount the host's Docker socket.
+		// Official document: https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/docker.md
+		"source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind"
+	],
+	"runArgs": [
+		"--name=gatling-operator-dev-container",
+		"--hostname=gatling-operator-dev-container",
+		// Set network mode to host to communicate with other containers.
+		"--network=host"
```
Comment (R-HNF) [inline .devcontainer/devcontainer.json:20]: It's necessary to control kind clusters on the host.

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	],
+	"containerEnv": {
+		"IN_DEV_CONTAINER": "true"
+	},
+	// Restore the local kubectl config to a dev container.
+	"postStartCommand": "if [ -d ${containerWorkspaceFolder}/.kube ]; then cp -r ${containerWorkspaceFolder}/.kube $HOME/.kube; fi",
```
Comment (R-HNF) [inline .devcontainer/devcontainer.json:26]: restoring the backed-up kubeconfig.

File: Dockerfile [hunk 4]
```diff
@@ -16,7 +16,7 @@ COPY controllers/ controllers/
 COPY pkg/ pkg/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=$(dpkg --print-architecture) go build -a -o manager main.go
```
Comment (R-HNF) [inline Dockerfile:19]: make it available for use in other architectures as well.

File: Makefile (hunk 1 above)
Comment (R-HNF) [reply Makefile]: `cp` command creates a `.kube` directory under the existing `.kube` directory, so I changed it to use `kind get kubeconfig`.

**Comment by R-HNF** [comment]
@itiB 
Please confirm that I have added how to use the dev container without using VSCode. Thanks:+1:

**Review by itiB — APPROVED** [review]

**Pull request merged**